pysite/views/main/jams/retract.py

```diff
@@ -0,0 +1,84 @@
+from werkzeug.exceptions import BadRequest
+
+from pysite.base_route import RouteView
+from pysite.decorators import csrf
+from pysite.mixins import DBMixin, OAuthMixin
+
+BANNABLE_STATES = ["preparing", "running"]
+
+
+class JamsProfileView(RouteView, DBMixin, OAuthMixin):
+    path = "/jams/retract"
+    name = "jams.retract"
+
+    table_name = "code_jam_participants"
+    infractions_table = "code_jam_infractions"
+    jams_table = "code_jams"
+    responses_table = "code_jam_responses"
+
+    def get(self):
+        if not self.user_data:
+            return self.redirect_login()
+
+        user_id = self.user_data["user_id"]
+        participant = self.db.get(self.table_name, user_id)
+
+        banned = False
+
+        if participant:
+            responses = self.db.run(self.db.query(self.responses_table).filter({"snowflake": user_id}), coerce=list)
+
+            for response in responses:
+                jam = response["jam"]
+                print(jam)
```

Comment:
this looks like a leftover from debugging?

Reply:
Done.